src/data/polynomial.lean

```diff
@@ -625,6 +625,19 @@ lemma degree_ne_of_nat_degree_ne {n : ℕ} :
   (λ _ h, option.no_confusion h)
   (λ n' h, mt option.some_inj.mp h)
 
+lemma coeff_nat_degree_eq_zero_iff {f : polynomial R} : f.coeff f.nat_degree = 0 ↔ f = 0 :=
```

Comment:
f.coeff f.nat_degree is called `leading_coeff`, and I think this lemma is already proved